### PR TITLE
Add a tsan runtime

### PR DESCRIPTION
**Pull request by fabbing**
# TODO

- Call `__tsan_func_exit` as appropriate when raising/catching exception
- Call  `__tsan_func_exit` when `perform`-ing effect and `__tsan_func_entry` when `resume`-ing
- Handle the `TODO` and `FIXME`

### CODE REVIEW

**Comment by OlivierNicole** [comment]
Now that #2 has been merged, could you remove its commit from this PR?

File: configure.ac [hunk 1]
```diff
@@ -366,6 +379,10 @@ AC_ARG_ENABLE([force-instrumented-runtime],
   [AS_HELP_STRING([--force-instrumented-runtime],
     [force the usage of the instrumented runtime])])
 
+AC_ARG_ENABLE([force-tsan-runtime],
+  [AS_HELP_STRING([--force-tsan-runtime],
+    [force the usage of the tsan runtime])])
+
```
Comment (OlivierNicole) [inline configure.ac]: What's the intended use for that flag? It is not used for now.

**Review by OlivierNicole — COMMENTED** [review]
The “Add a tsan runtime” commit looks good to me except for one thing (see review question), we just need to solve the TODOs and FIXMEs. The other commits belong to another PR, I think.

File: configure.ac (hunk 1 above)
Comment (fabbing) [reply configure.ac]: It's used to force to link with a the instrumented/tsan runtime but we deemed it unnecessary (we will use the `-tsan` flag instead).

**Comment by OlivierNicole** [comment]
This looks good I think. Weird that there is one macOS test failing, seemingly completely unrelated to your changes… I launched a re-run just in case it's flaky.

For future reference, would it be possible to clarify the description of this PR above? (I think the last two list items can be removed.)

**Comment by OlivierNicole** [comment]
> Weird that there is one macOS test failing, seemingly completely unrelated to your changes… I launched a re-run just in case it's flaky.

Still failing. Well, it's better than the current situation where macOS does not build at all, so I say we merge this and I'll open a follow-up issue for this test.

**Comment by OlivierNicole** [comment]
I took the liberty of squashing in some of the “make the linter happy” commits.

**Pull request merged**